pkg/queue/manager.go

```diff
@@ -174,7 +179,14 @@ func (m *Manager) AddClusterQueue(ctx context.Context, cq *kueue.ClusterQueue) e
 		return errClusterQueueAlreadyExists
 	}
 
-	cqImpl, err := newClusterQueue(ctx, m.client, cq, m.workloadOrdering, m.admissionFairSharingConfig)
+	var penalties map[string]corev1.ResourceList
+	if features.Enabled(features.AdmissionFairSharing) {
+		m.penaltyMap.RLock()
+		defer m.penaltyMap.RUnlock()
+
+		penalties = m.penaltyMap.penalties
+	}
```

Comment:
Please wrap this as a function on `penaltyMap.Snapshot()` which gives you the snapshot of penalties.

Reply:
So we can then read from the updated version